src/luascript.cpp

```diff
@@ -4256,6 +4338,33 @@ int LuaScriptInterface::luaSendGuildChannelMessage(lua_State* L)
 	return 1;
 }
 
+
+int LuaScriptInterface::luaLoadDirectory(lua_State* L)
+{
+	const boost::filesystem::path& dir = boost::filesystem::current_path().string() + getString(L, 1);
+
+	if (!boost::filesystem::exists(dir) || !boost::filesystem::is_directory(dir)) {
+		pushBoolean(L, false);
+		return 1;
+	}
+
+	boost::filesystem::recursive_directory_iterator it(dir);
+	boost::filesystem::recursive_directory_iterator endit;
+
+	for (; it != endit; ++it) {
+		if (is_regular_file(*it) && it->path().extension() == ".lua") {
```

Comment:
Either specify namespaces or rely on ADL. Don't mix them both because it's confusing.